email/exporter.go

```diff
@@ -0,0 +1,161 @@
+package email
+
+import (
+	"context"
+	"errors"
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"golang.org/x/time/rate"
+	"google.golang.org/protobuf/types/known/emptypb"
+
+	"github.com/letsencrypt/boulder/core"
+	emailpb "github.com/letsencrypt/boulder/email/proto"
+	berrors "github.com/letsencrypt/boulder/errors"
+	blog "github.com/letsencrypt/boulder/log"
+)
+
+const (
+	// five is the number of concurrent workers processing the email queue. This
+	// number was chosen specifically to match the number of concurrent
+	// connections allowed by the Pardot API.
+	five = 5
```

Comment:
This feels like it should be in pardot.go, alongside contactsPath and other pardot-specific constants.

Reply:
I don't quite agree with this, but if you feel strongly about it I'm happy to move it.

Reply:
I'm now coming around to the idea that it shouldn't live in pardot.go, but instead should be named `maxConcurrentRequests` and should be _configurable_.

This is because, if we run two copies of the exporter (one in each DC), they'll likely step on each other's toes and surpass pardot's concurrent-connections limit. SRE may have a vested interest in limiting each instance to just 2 or 3 workers, not 5.

Reply:
Good catch here!